connector/countconnector/connector.go

```diff
@@ -0,0 +1,125 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package countconnector // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/countconnector"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const scopeName = "otelcol/countconnector"
+
+// count can count spans, data points, or log records and emit
+// the count onto a metrics pipeline.
+type count struct {
+	Config
+	consumer.Metrics
+	component.StartFunc
+	component.ShutdownFunc
+}
+
+func (c *count) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
```

Comment:
Doesn't a connector converting from one signal to another inherently mutates the data?

Reply:
I thought that the mutating flag was for the original data set. Do connectors mess with the original telemetry?

Reply:
I think the guidelines for this need to be clarified yet but if I understand correctly the purpose of `MutatesData` is to manage shared data appropriately. Specifically, to make copies for components that will mutate the data, but otherwise just share the original struct.

In this case, the struct is read and dropped, but not altered - another component with a handle to the same struct would not be affected.

In theory, we could have connectors that mutate data but I don't currently believe this will be common. An example I can imagine would be a connector that handles multiple data types simultaneously and enriches data based on multi-signal logic. e.g. When an ERROR log is observed, generate an event ID and for the next minute, annotate this ID onto all data from the associated resource.